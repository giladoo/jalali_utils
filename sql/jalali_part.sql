CREATE EXTENSION jalali_utils;

SELECT jalali_part('year', '2019-07-07 14:10:52.84937+04:30');

SELECT jalali_part('month', '2019-07-07 14:10:52.84937+04:30');

SELECT jalali_part('day', '2019-07-07 14:10:52.84937+04:30');

SELECT jalali_part('hour', '2019-07-07 14:10:52.84937+04:30');

SELECT jalali_part('minute', '2019-07-07 14:10:52.84937+04:30');

SELECT jalali_part('second', '2019-07-07 14:10:52.84937+04:30');

SELECT jalali_part('doy', '2019-07-07'); -- 1398/04/16

SELECT jalali_part('doy', '2019-09-22'); -- 1398/06/31

SELECT jalali_part('doy', '2020-03-19'); -- 1398/12/29

SELECT jalali_part('doy', '2021-03-20'); -- 1399/12/30

SELECT jalali_part('doy', '2021-03-21'); -- 1400/01/01

SELECT jalali_part('quarter', '2021-03-21');

SELECT jalali_part('quarter', '2021-03-20');

SELECT jalali_part('decade', '2021-03-21');

SELECT jalali_part('decade', '2021-03-20');

SELECT jalali_part('century', '2020-03-20');

SELECT jalali_part('century', '2022-03-20');

SELECT jalali_part('century', '2022-03-21');

SELECT jalali_part('dow', '2001-10-16');

SELECT jalali_part('dow', '2020-10-17');

SELECT jalali_part('dow', '2020-10-23');

DROP EXTENSION jalali_utils;
