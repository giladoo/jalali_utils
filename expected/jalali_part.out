CREATE EXTENSION jalali_utils;
SELECT jalali_part('year', '2019-07-07 14:10:52.84937+04:30');
 jalali_part 
-------------
        1398
(1 row)

SELECT jalali_part('month', '2019-07-07 14:10:52.84937+04:30');
 jalali_part 
-------------
           4
(1 row)

SELECT jalali_part('day', '2019-07-07 14:10:52.84937+04:30');
 jalali_part 
-------------
          16
(1 row)

SELECT jalali_part('hour', '2019-07-07 14:10:52.84937+04:30');
 jalali_part 
-------------
          14
(1 row)

SELECT jalali_part('minute', '2019-07-07 14:10:52.84937+04:30');
 jalali_part 
-------------
          10
(1 row)

SELECT jalali_part('second', '2019-07-07 14:10:52.84937+04:30');
 jalali_part 
-------------
          52
(1 row)

SELECT jalali_part('doy', '2019-07-07'); -- 1398/04/16
 jalali_part 
-------------
         109
(1 row)

SELECT jalali_part('doy', '2019-09-22'); -- 1398/06/31
 jalali_part 
-------------
         186
(1 row)

SELECT jalali_part('doy', '2020-03-19'); -- 1398/12/29
 jalali_part 
-------------
         365
(1 row)

SELECT jalali_part('doy', '2021-03-20'); -- 1399/12/30
 jalali_part 
-------------
         366
(1 row)

SELECT jalali_part('doy', '2021-03-21'); -- 1400/01/01
 jalali_part 
-------------
           1
(1 row)

SELECT jalali_part('quarter', '2021-03-21');
 jalali_part 
-------------
           1
(1 row)

SELECT jalali_part('quarter', '2021-03-20');
 jalali_part 
-------------
           4
(1 row)

SELECT jalali_part('decade', '2021-03-21');
 jalali_part 
-------------
        1400
(1 row)

SELECT jalali_part('decade', '2021-03-20');
 jalali_part 
-------------
        1390
(1 row)

SELECT jalali_part('century', '2020-03-20');
 jalali_part 
-------------
          14
(1 row)

SELECT jalali_part('century', '2022-03-20');
 jalali_part 
-------------
          14
(1 row)

SELECT jalali_part('century', '2022-03-21');
 jalali_part 
-------------
          15
(1 row)

SELECT jalali_part('dow', '2001-10-16');
 jalali_part 
-------------
           3
(1 row)

SELECT jalali_part('dow', '2020-10-17');
 jalali_part 
-------------
           0
(1 row)

SELECT jalali_part('dow', '2020-10-23');
 jalali_part 
-------------
           6
(1 row)

DROP EXTENSION jalali_utils;
